### ISSUE TITLE
Potential memory leaks

### ISSUE THREAD

**Issue by Quantomicus** [DivanteLtd/vue-storefront#2173]
## Current behavior
I have observed a constant rise in memory usage of vue-storefront while it has running. 

## Expected behavior
For vue-storefront not to constantly eat up more and more memory on the server.

## Steps to reproduce the issue

1. Run vue-storefront (and vue-storefront-api)
2. Leave it running for a longer period of time
3. Observe results

![image](https://user-images.githubusercontent.com/9385129/51084516-6eb68a00-172b-11e9-9754-d5262d54c85c.png)
_Started off the previous day on reboot, observed consistent rise from around 5% memory usage (~33% on the graph total) to 28% memory usage (~80% on graph total) at the time of the screenshot (server has 8GB of RAM)_

## Repository
Unknown, could be either vue-storefront or vue-storefront-api.

# Can you handle fixing this bug by yourself?
NO

## Environment details
- Browser: irrelevant
- OS: CentOS 7
- Node: v10.15.0

## Additional information
If this is expected behavior, I will close this issue, but this doesn't seem sustainable. I will gladly provide any additional diagnostics information.

**Comment by pkarw** [comment]
Hi, thanks! This observation refers to `vue-storefront` either `vue-storefront-api`?

@tomasz-duda has already optimized the memory usage for `vue-storefront`. We probably need to dig a little deepr :)

If this problem is significant enough please set the `ecosystem.json` limit for memory usage in `pm2` + please turn on the `SSR cache` (more on this in our [docs](https://docs.vuestorefront.io/guide/basics/ssr-cache.html))

**Comment by Quantomicus** [comment]
It appears to be related to `vue-storefront`. I have now enabled SSR cache and will report back with results after some time has passed.

**Comment by filrak** [comment]
@Quantomicus is this a develop branch?

**Comment by pkarw** [comment]
@filrak good point! @Quantomicus if this is develop - then would be great to get comparison data on Your env with the master; I mean - we've introduced lazy loading in the current develop and I'm just wondeirng if it could have caused this kind of issues (probably not)

**Comment by Quantomicus** [comment]
This is not develop, no. Sorry :(

**Comment by Quantomicus** [comment]
After some hours of running with SSR cache enabled, the rise in memory usage seems to be more gradual and the initial memory use is smaller as well, but there is still a consistent gradual rise of memory usage. I am confident I could stop this by limiting the memory usage, but I don't need to do that at the moment and I feel it is still worth it to investigate deeper.

**Comment by pkarw** [comment]
Ok, well try to profile it by the next chance

**Comment by Quantomicus** [comment]
I could use some instructions on how to do that, I found nothing related in the docs. Maybe I'm just not looking hard enough :)

**Comment by Quantomicus** [comment]
The issue has indeed been fixed, thank you :)

**Comment by pkarw** [comment]
Great to hear that :-)
